packages/celotool/src/lib/geth.ts

```diff
@@ -1135,9 +1144,12 @@ export async function connectPeers(instances: GethInstanceConfig[], verbose: boo
   )
 }
 
-// Add validator 0 as a peer of each other validator.
+// Peer every validator so they are fully connected
 export async function connectValidatorPeers(instances: GethInstanceConfig[]) {
   await connectPeers(
-    instances.filter(({ wsport, rpcport, validating }) => validating && (wsport || rpcport))
+    instances.filter(
+      ({ wsport, rpcport, validating, isProxy, isProxied }) =>
+        isProxy || (validating && !isProxied && (wsport || rpcport))
```

Comment:
Parenthesis are a bit off unless I am mistaken.
```suggestion
        (isProxy || (validating && !isProxied)) && (wsport || rpcport)
```

Reply:
Good catch!